geth/node/manager.go

```diff
@@ -563,3 +565,16 @@ func (m *NodeManager) RPCServer() (*rpc.Server, error) {
 
 	return m.rpcServer, nil
 }
+
+// isNodeAvailable check if we have a node running and make sure is fully started
+func (m *NodeManager) isNodeAvailable() error {
+	if m.nodeStarted == nil {
+		return ErrNoRunningNode
+	}
+
+	if m.node == nil {
+		return ErrNodeOffline
+	}
```

Comment:
Probably, the check for starting node should be before checking its `nodeStarted` value. However, I'm not a proponent for such kind of defensive programming.

Reply:
Actually, `m.nodeStarted` is set before `m.node`.

Reply:
You're right. Was too sleepy and thought that check was `m == nil` somehow %)